neuromation/api/file_filter.py

```diff
@@ -24,3 +27,76 @@ def include(self, pattern: str) -> None:
             if result == exclude and matcher(path):
                 result = not result
         return result
+
+
+def translate(pat: str) -> str:
+    """Translate a shell PATTERN to a regular expression.
+    """
+
+    i = 0
+    n = len(pat)
+    res = ""
+    while i < n:
+        c = pat[i]
+        i += 1
+        if c == "*":
+            if (
+                (not res or res[-1] == "/")
+                and i < n
+                and pat[i] == "*"
+                and (i + 1 == n or pat[i + 1] == "/")
+            ):
+                # ** between slashes or ends of the pattern
+                if i + 1 == n:
+                    res += ".*"
+                else:
+                    res += "(?:.+/)?"
+                i += 2
+            else:
+                # Any other *
+                res += "[^/]*"
+        elif c == "?":
+            res += "[^/]"
+        elif c == "/":
```

Comment:
Special case for `/` was added just to make tests easier: in 3.6 `/` is escaped by `re.escape()`, in 3.7 it is not escaped.